src/position.cpp

```diff
@@ -787,7 +787,7 @@ void Position::do_move(Move m, StateInfo& newSt, bool givesCheck) {
       if (   (int(to) ^ int(from)) == 16
           && (attacks_from<PAWN>(to - pawn_push(us), us) & pieces(them, PAWN)))
       {
-          st->epSquare = (from + to) / 2;
+          st->epSquare = Square((int(from) + int(to)) / 2);
```

Comment:
In general I like this patch but just want to make a small tangential comment. Instead of `(from + to) / 2` you could use `to - pawn_push(us)` like on the previous line. Or, if the trick `to ^ from == 16` is acceptable on line 787 I wonder if the trick `to ^ 8` is acceptable on lines 788 and 790. I think it's correct... xoring with 8 swaps ranks 1 and 2, ranks 3 and 4, ranks 5 and 6 etc...

Reply:
I agree that "to - pawn_push(us)" looks better and makes more sense here (and has been computed anyway at this point). 

And I guess "Square(to ^ 8)" compiles to fewer machine instructions than either "to - pawn_push(us)" or "(from + to)/2", but Marco might be happier with the saner looking "to - pawn_push(us)".